homeassistant/components/blink/__init__.py

```diff
@@ -1,157 +1,153 @@
                 vol.Required(CONF_PASSWORD): cv.string,
-                vol.Optional(
-                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
-                ): cv.time_period,
-                vol.Optional(CONF_BINARY_SENSORS, default={}): BINARY_SENSOR_SCHEMA,
-                vol.Optional(CONF_SENSORS, default={}): SENSOR_SCHEMA,
-                vol.Optional(CONF_OFFSET, default=1): int,
-                vol.Optional(CONF_MODE, default=""): cv.string,
+                vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): int,
             }
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-def setup(hass, config):
-    """Set up Blink System."""
-
-    conf = config[BLINK_DATA]
-    username = conf[CONF_USERNAME]
-    password = conf[CONF_PASSWORD]
-    scan_interval = conf[CONF_SCAN_INTERVAL]
-    is_legacy = bool(conf[CONF_MODE] == "legacy")
-    motion_interval = conf[CONF_OFFSET]
-    hass.data[BLINK_DATA] = blinkpy.Blink(
-        username=username,
-        password=password,
-        motion_interval=motion_interval,
-        legacy_subdomain=is_legacy,
-    )
-    hass.data[BLINK_DATA].refresh_rate = scan_interval.total_seconds()
-    hass.data[BLINK_DATA].start()
+async def async_setup(hass, config):
+    """Set up a config entry."""
+    if DOMAIN not in config:
+        return True
 
-    platforms = [
-        ("alarm_control_panel", {}),
-        ("binary_sensor", conf[CONF_BINARY_SENSORS]),
-        ("camera", {}),
-        ("sensor", conf[CONF_SENSORS]),
-    ]
+    conf = config.get(DOMAIN, {})
 
-    for component, schema in platforms:
-        discovery.load_platform(hass, component, DOMAIN, schema, config)
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Blink via config entry."""
+    if DOMAIN not in hass.data:
+        username = entry.data[CONF_USERNAME]
+        password = entry.data[CONF_PASSWORD]
+        scan_interval = entry.data[CONF_SCAN_INTERVAL]
+        blink = blinkpy.Blink(
+            username=username,
+            password=password,
+            motion_interval=DEFAULT_OFFSET,
+            legacy_subdomain=False,
+            no_prompt=True,
+            device_id="Home Assistant",
+        )
+        blink.refresh_rate = scan_interval
+        await hass.async_add_executor_job(blink.start)
+        hass.data[DOMAIN] = {entry.unique_id: blink}
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
 
-    def trigger_camera(call):
+    def force_refresh():
+        """Force a blink refresh."""
+        return hass.data[DOMAIN][entry.unique_id].refresh(force_cache=True)
```

Comment:
`force_cache` seems like it does the opposite of what a function called `force_refresh` would do.